joomla/joomla-cms
# [3.10] [WiP] Add postinstall message for invalid created date check

Pull Request for Issue #30546 (part).

Related to Pull Request #31449 .

### Summary of Changes

WiP - work in progress

With pull request (PR) #31449 , the update SQL scripts of Joomla 4 will be extended by setting a valid default date value (January 1, 1980 00:00:00 UTC) for the created date of core content which has an old style pseudo null date as created date, in order to fix issue #30546 .

This PR here adds a new postinstall message to Joomla 3 to inform the admin about this and so enable him or her to correct bad created dates to a different value than the default mentioned above.

The message is only shown if such bad created date values have been found in database.

If the admin wants to set custom created dates (e.g. the date when the site was created would make sense), it should be done latest before updating to Joomla 4.

But it makes sense to show the postinstallation message already in 3.9.x and not only in 3.10 so the admins can start early with fixing their created dates.

**_To be done_** Add a link to a docs page (also to be done) with the necessary SQL statements to find out which records of which kind of content have the bad old null date value for created dates.

### Testing Instructions

This pull request should be tested with different database types (MySQL or MariaDB and PostgreSQL, and theoretically also MS SQL Server).

Testers please report back which kind of database you have used for testing. If you have more than one type, test with all you have. Thanks in advance.

1. Have a Joomla 3.9.x installation, either current staging branch or latest 3.9.x nightly or 3.9.22, which has some content of each of the following types:
- Articles
- Banners
- Categories
- Contacts
- News Feeds
- Tags
- User Notes

If you don't have such content, either create some for each type or make a new installation and chose to install "Test English (GB) Sample Data" at the final step and then create a user note in backend.

2. Use a database client like e.g. phpMyAdmin for a MySQL or MariaDB database or phpPgAdmin for a PostgreSQL database to set an old pseudo null date as created date for some content.

_(to be completed)_

### Actual result BEFORE applying this Pull Request

There is no postinstall message about invalid created dates in database.

### Expected result AFTER applying this Pull Request

If some core content contains an old style null date as created date, the following postinstall message is shown:

![j3-postinstall-message-invalid-created-date](https://user-images.githubusercontent.com/7413183/99875877-18ea7d00-2bf3-11eb-8099-7ba7497af22d.png)

The message lists all types of content which could be affected, i.e. not only those where the bad created date was found.

To be done: I have to add a link to a doc which provides the necessary SQL statements to find the affected records for each content type.

If no content was found with a bad created date, the postinstall message is not shown.

### Documentation Changes Required

It needs some doc which provides the necessary SQL statements to find the affected records for each content type.
